### PR TITLE
Refactor of Primitives Classes to Return Object3d objects

### PR DESCRIPTION
**Pull request by voss70**
This is a major refactor of the way we manage geometric primitives in the modeler. All create_xxx functions of the Primitives modules return objects rather than ids. This may break some existing applications. All examples in our suite have been adjusted.

We also introduce unit-testing from IronPython using unittest. This is a preliminary implementation and will be extended in the near future.

### CODE REVIEW

**Review by maxcapodi78 — CHANGES_REQUESTED** [review]

File: pyaedt/modeler/Modeler.py [hunk 1]
```diff
@@ -866,6 +866,7 @@ def add_workbench_link(self, objects, ambient_temp=22, create_project_var=False,
         vdef = []
         for obj in objects:
             mat = self.primitives[obj].material_name
+            print("Material" + mat)
```
Comment (akaszynski) [inline pyaedt/modeler/Modeler.py:869]: Let's avoid adding print statements and instead use the standard python logging module.  Users won't want their stdout polluted.

**Comment by akaszynski** [comment]
Please merge with `main` so the new CI/CD can run.

File: pyaedt/modeler/Modeler.py (hunk 1 above)
Comment (voss70) [reply pyaedt/modeler/Modeler.py:869]: Fair enough ...

**Comment by voss70** [comment]
> Please merge with `main` so the new CI/CD can run.

Still working on the pull request. I expect to merge in the next day or two

**Comment by akaszynski** [comment]
> Still working on the pull request. I expect to merge in the next day or two

Please consider converting this PR to a draft so you can signal when it's ready for review.

**Comment by voss70** [comment]
> > Still working on the pull request. I expect to merge in the next day or two
> 
> Please consider converting this PR to a draft so you can signal when it's ready for review.

That feels better 👍

File: pyaedt/modeler/Modeler.py (hunk 1 above)
Comment (voss70) [reply pyaedt/modeler/Modeler.py:869]: I'll address this in a complete way in a coming PR. Requesting to leave this for now. I'll open an issue to keep track of this

**Comment by voss70** [comment]
@akaszynski @maxcapodi78 is there any way to avoid the new test cases planned for our IronPython unit test (in _unittest_ironpython/ ) to be run in our pytest ? These are causing issues at the moment.

**Comment by voss70** [comment]
@maxcapodi78 I need your help regarding the 4 failing unit tests. Thx

**Comment by maxcapodi78** [comment]
> @maxcapodi78 I need your help regarding the 4 failing unit tests. Thx

yes, we can work on that tomorrow

**Comment by maxcapodi78** [comment]
> @akaszynski @maxcapodi78 is there any way to avoid the new test cases planned for our IronPython unit test (in _unittest_ironpython/ ) to be run in our pytest ? These are causing issues at the moment.  


I think we need to slightly modify the pipeline to skip ironphyton folder

**Review by maxcapodi78 — APPROVED** [review]

**Comment by voss70** [comment]
Great to get this one done !

**Pull request merged**